juju/juju
# Merge 2.9

Merge 2.9

#13689 [JUJU-416] Consistantly use juju/retry to handle retries 7 (provider/maas/*)
#13633 [JUJU-478] Retry ping before failing
#13662 [JUJU-474] Create and save a display name for bootstrap machine on oci
#13702 Add CharmDownloader to allowed caas facades
#13704 Removing commented out code left behind.
#13685 [JUJU-503] Reimplement essential metadata for charmhub using refresh API
#13708 [JUJU-575] Add trust for kubeflow deploy command;
#13707 [JUJU-576] Update upstream deps, including using txn/v2 and worker/v3
#13709 Use str option value for --trust and pass --trust to juju

Conflicts were mainly import paths.
There was a bigger issue with the maas provider due to deleted v1 code.
```
# Conflicts:
#       apiserver/facades/client/modelmanager/modelmanager.go
#       apiserver/restrict_caasmodel.go
#       cloudconfig/instancecfg/instancecfg.go
#       core/charm/repository/charmhub_test.go
#       go.mod
#       go.sum
#       provider/maas/environ.go
#       provider/maas/environ_test.go
#       provider/maas/environ_whitebox_test.go
#       scripts/juju-blobstore-cleanup/main.go
#       scripts/win-installer/setup.iss
#       snap/snapcraft.yaml
#       state/action.go
#       state/action_test.go
#       state/restore.go
#       state/restore_test.go
#       version/version.go
```
## QA steps

See PRs



[JUJU-416]: https://warthogs.atlassian.net/browse/JUJU-416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-478]: https://warthogs.atlassian.net/browse/JUJU-478?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-474]: https://warthogs.atlassian.net/browse/JUJU-474?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-503]: https://warthogs.atlassian.net/browse/JUJU-503?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-575]: https://warthogs.atlassian.net/browse/JUJU-575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[JUJU-576]: https://warthogs.atlassian.net/browse/JUJU-576?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
